tests/roundtrip.rs

```diff
@@ -172,6 +173,29 @@ impl TestState {
             .context("failed to validate the `print` snapshot")?;
         self.bump_ntests();
 
+        let mut test_folding = true;
+        for part in test.iter().filter_map(|t| t.to_str()) {
+            match part {
+                "legacy-exceptions" => test_folding = false,
+                _ => (),
+            }
+        }
+
+        let mut folded_string = String::new();
+        if test_folding {
+            // Same test, printing instructions in folded form.
+            let mut folding_printer = wasmprinter::Config::new();
+            folding_printer.fold_instructions(true);
+            folding_printer
+                .print(contents, &mut PrintFmtWrite(&mut folded_string))
+                .context("failed to print wasm in folded form")?;
+            self.bump_ntests();
+            // Snapshot the folded WAT as well.
+            self.snapshot("print-folded", test, &folded_string)
+                .context("failed to validate the `print-folded` snapshot")?;
+            self.bump_ntests();
+        }
```

Comment:
Question for you, how useful was this during development? Do you think it's worthwhile to have both the normal and folded forms checked in under `snapshots`? I occasionally second-guess having `snapshots` for all the spec tests since they're so large and this is definitely increasing the size, but I wanted to see if you found this useful during development and whether it helped uncover bugs.

The alternative to this would be to add tests under `tests/cli/*.wat` for various forms of folded instructions (e.g. the `fold-multi.wat` added above). That has much less test coverage though.

Another possible alternative though, which I actually think is probably a good idea regardless, is that it'd be possible to hook things up to the fuzzers where `fuzz/src/print.rs` could get the ability to toggle the folded option and `fuzz/src/validate_valid_module.rs` could perform a number of the same assertion checks that this test module does over there as well.

Well in any case no need to handle all the fuzzing bits here, I can also tackle that later. For now though I'm mostly curious how useful you found these were during development.

Reply:
No strong feelings. It was helpful when developing the feature, since I could see if I was changing anything (or what I was changing)... but not sure it needs to be committed upstream for all time, vs. a smaller selection of folding examples as you propose. Maybe the nice thing would be to be able to do a regression test privately, where you give two Git hashes and it tells you what if anything has changed across them (running the whole testsuite), without checking it all in to the repo.

I do think it's very nice to be checking the binary roundtrippability of the folded text format across the whole testsuite (even if it doesn't check the folded text against a snapshot).

Reply:
Ok how about this:

* Enhance `roundtrip.rs` to ensure that round-tripping through the text format tests both the folded and non-folded forms.
* Run the "is whitespace pretty" check for both folded and non-folded forms.
* Add selective tests to `tests/cli/*.wat` as necessary.
* Leave out folded versions of `snapshot` tests for now.
* (future PR) add support to the fuzzer for testing round-tripping with folded forms

That way we get a strong level of baseline support for correctness and it's just stylistic bits that may fall through the cracks?

Reply:
Done (minus the fuzzing support).